modules/ROOT/pages/bolt/handshake.adoc

```diff
@@ -91,6 +92,103 @@ S: 00 00 00 00
 ----
 
 
+[[bolt-version57]]
+== Bolt version 5.7
+With bolt version 5.7, a new scheme for handshaking is introduced: **Manifest v1**.
+
+[[varint]]
+=== VarInt
+VarInts are transmitted in groups of 7 bits with their least significant bits first.
+The most significant bit within each byte is used as a continuation bit, indicating that more data is to follow.
+As such, it is not to be considered part of the decoded value.
+
+[cols="1,1,1",options="header"]
+[%autowidth]
+|===
+| Hex Encoded
+| Binary Decoded
+| Decimal Decoded
+
+| `01`
+| `0000 0001`
+| `1`
+
+| `7F`
+| `0111 1111`
+| `127`
+
+| `FF 82 71`
+| `0111 1111`
+| `127`
+|===
+
+Let's take apart the last example:
+
+----
+# Encoded in hexadecimal
+FF 82 71
+# Encoded in binary
+1111 1111  1000 0010  0111 0001
+# These are the continuation bits specifying how many bytes to read
+1... ....  1... ....  0... ....
+# Bits of the actual value (in 7-bit groups in little-endian order)
+ 111 1111   000 0010   111 0001
+ ========   ^^^^^^^^   ~~~~~~~~
+# Convert to big-endian (if target machine is big-endian)
+ 111 0001   000 0010   111 1111
+ ~~~~~~~~   ^^^^^^^^   ========
+# Concatenate bits for the final (big-endian) binary representation
+0001 1100  0100 0001  0111 1111
+   ~~~~~~~~~~^^^^^^^^^^========
```

Comment:
It's not clear how we came from `0001 1100  0100 0001  0111 1111` to `127`. How does the continuation bit plays here?!

Reply:
The continuation bits play no longer any role at all. We stripped it in step 4 (called "its of the actual value").

I'm not sure how I can make that clearer. Do you have a suggestion?

Reply:
OH!! There's a copy pasta mistake :runner::dash:

Reply:
Fixed. Does it make sense to you now?